gwt-localizable-processor/src/main/java/org/gwtproject/i18n/processor/LocalizableProcessingStep.java

```diff
@@ -47,105 +55,378 @@ public LocalizableProcessingStep(ProcessingEnvironment processingEnv) {
+                                        TypeName.get(constantMethod.getMethod().getReturnType()))
+                                    .addCode(codeBlock)
+                                    .build());
+                          });
+
+                    } else {
+                      constantImplBuilder.addMethod(
+                          MethodSpec.methodBuilder(
+                                  constantMethod.getMethod().getSimpleName().toString())
+                              .addModifiers(Modifier.PUBLIC)
+                              .addAnnotation(Override.class)
+                              .returns(TypeName.get(constantMethod.getMethod().getReturnType()))
+                              .addCode(
+                                  getValueExpression(
+                                      context, constantMethod, locale, localeResourceMap))
+                              .build());
+                    }
+                  });
+
+          if (context.isAssignableFrom(element, ConstantsWithLookup.class)) {
+            lookupMethodsBuilder
+                .lookupMethods()
+                .forEach(methodBuilder -> constantImplBuilder.addMethod(methodBuilder.build()));
+          }
+
+          writeSource(constantImplBuilder, context, element);
+        });
+  }
+
+  private String getConstantClassName(Element element, String locale) {
+    String enclosingElementName =
+        !(ElementKind.PACKAGE == element.getEnclosingElement().getKind())
+            ? element.getEnclosingElement().getSimpleName().toString() + "_"
+            : "";
+    return enclosingElementName + element.getSimpleName().toString() + "_" + localePostfix(locale);
+  }
+
+  private String localePostfix(String locale) {
+    return "default".equals(locale) ? "" : locale;
+  }
+
+  private void addCacheField(TypeSpec.Builder constantImplBuilder, boolean[] cacheAdded) {
+    if (!cacheAdded[0]) {
+      constantImplBuilder.addField(
+          FieldSpec.builder(ClassName.get(Map.class), "cache")
+              .initializer("new $T()", HashMap.class)
+              .build());
+      cacheAdded[0] = true;
+    }
+  }
+
+  private Optional<CodeBlock> getMapMethodBody(
+      ConstantMethod method, String locale, Map<String, Properties> localeResourceMap) {
+    CodeBlock.Builder builder = CodeBlock.builder();
+
+    String key = method.getKey();
+    String[] property = method.getArrayPropertyValue(locale, key, localeResourceMap);
+    if (nonNull(property)) {
+      builder.addStatement(
+          "$T args = ($T)cache.get($S)",
+          ParameterizedTypeName.get(Map.class, String.class, String.class),
+          ParameterizedTypeName.get(Map.class, String.class, String.class),
+          key);
+      builder
+          .beginControlFlow("if(args == null)")
+          .addStatement(
+              "args = new $T()",
+              ParameterizedTypeName.get(LinkedHashMap.class, String.class, String.class));
+
+      if (property.length > 0) {
```

Comment:
if the array length is zero, the stream won't do anything, so we don't need this check